modules/fundamental/src/ai/service/tools/package_info.rs

```diff
@@ -67,20 +67,14 @@ Input: The package name, its Identifier URI, or UUID.
             .ok_or("Input should be a string")?
             .to_string();
 
-        // Try lookup as a PURL
-        let mut purl_details = match Purl::try_from(input.clone()) {
+        let mut purl_details = match service
+            .fetch_purl_details(&[input.as_str()], Deprecation::Ignore, db)
+            .await
+        {
             Err(_) => None,
-            Ok(purl) => service.purl_by_purl(&purl, Deprecation::Ignore, db).await?,
+            Ok(details) => details.get(&input.clone()).cloned(),
```

Comment:
I'm actually surprised clippy didn't catch this.
```suggestion
            Ok(details) => details.get(&input).cloned(),
```